src/sidebar/util/groups.js

```diff
@@ -65,6 +65,10 @@ export function combineGroups(userGroups, featuredGroups, uri, settings) {
   return groups;
 }
 
+/**
+ * @param {Group} group
+ * @param {string} uri
+ */
```

Comment:
I'm wondering how many of the `@return` docs we need to flush out or just not worry if TS does not complain?

Reply:
Providing you specify the types of the parameters, TS can infer the return type automatically. However there are reasons why you may want to specify it explicitly:

1. To help human readers
2. To get TS to check that the returned type is what it should be (for example, if a function is documented with `@return {Annotation}` and you have a code path that returns `null` or some other value that is not compatible with this, TS will complain, which is good because other code using that function needs to handle these other possibilities!)
3. In some cases, this can enable more readable/succinct error messages if two types don't match up (for example, if you call a function that returns an object and then try to access a property that doesn't exist on that object, TS will report "Cannot access property X on type <Name>" instead of "Cannot access property X on type { < list of all the properties that the function returns > }")

We'll need to figure out the right balance here, but my approach so far has been to omit the return type if the function is short and the return type is "obvious", but specify it otherwise.

Reply:
I have been noticing that error today and its more helpful to see the list of types when possible. I suppose we could do multiple passes and fill in return values.